community/cypher/cypher/src/test/scala/org/neo4j/cypher/internal/compatibility/v3_4/runtime/commands/expressions/StringFunctionsTest.scala

```diff
@@ -54,7 +54,7 @@ class StringFunctionsTest extends CypherFunSuite {
     left("hello", 8) should equal(stringValue("hello"))
     left(null, 8) should equal(expectedNull)
     intercept[CypherTypeException](left(1042, 2))
-    intercept[StringIndexOutOfBoundsException](left("hello", -4))
+    intercept[IndexOutOfBoundsException](left("hello", -4))
```

Comment:
Why are we changing this exception type? If we wanted to we should be able to use the same type as before, should we not?

Reply:
Here we are relying on exceptions from the underlying implementation, it used to be an error coming form `java.lang.String.substring` now it is an error coming from invalid access of an array. Both are equally bad in my opinion we should throw something we control instead of just passing through underlying errors, but since `StringIndexOutOfBoundsException` extends `IndexOutOfBoundsException` I didn't see that this at least didn't make things any worse, and we could postpone a real fix to later work?